### ISSUE TITLE
NVDA stops responding to keyboard input

### ISSUE THREAD

**Issue by nvaccessAuto** [nvaccess/nvda#2003]
**Reported by mehgcap on 2011-12-17 02:26**
This ticket will be rather vague, but I cannot come up with a way to reproduce the problem. What happens is, randomly, NVDA will either stop speaking altogether or will stop responding to certain keys. If the latter, it is not always the same keys (sometimes it is the arrows, sometimes tab) but it always fails to respond to commands using the NVDA key. (I am using a desktop, in case the layout matters.) For instance, I will go to the desktop and hit ins-q to exit NVDA, but Windows makes the sound made when a character is pressed that does not correspond to anything on the screen, and NVDA simply says "q". Clearly, it is not responding to the insert key as it should and Windows thinks I just hit q, not ins-q. I must therefore go into task manager and kill the NVDA process, and I sometimes get a warning from the NVDA helper remote that the thread is dead or not responding, or something along those lines. If I then restart NVDA using the ctrl-alt-n shortcut, it sometimes goes back to normal. Often, though, especially if speech had stopped completely, I have to restart the computer or NVDA will either not speak or will behave oddly as described above (not responding to some keys).

Below is the log file stored in %temp%\nvda.log after it would not respond to the insert key but was still speaking (though it would finish everything it was saying, not interrupting itself when a new task was switched to or an arrow key was pressed as it should have done). I had just killed NVDA with the task manager and opened this file before restarting NVDA with the keyboard shortcut.
Log follows:

```
INFO - nvda (07:53:37):
Starting NVDA
INFO - core.main (07:53:45):
Config dir: C:\Users\Alex\AppData\Roaming\nvda
INFO - core.main (07:54:01):
NVDA version 2011.3
INFO - core.main (07:54:01):
Using Windows version sys.getwindowsversion(major=6, minor=1, build=7600, platform=2, service_pack='')
INFO - core.main (07:54:01):
Using Python version 2.7.2 (default, Jun 12 2011, 15:08:59) [v.1500 32 bit (Intel)](MSC)
INFO - core.main (07:54:01):
Using comtypes version 0.6.2
INFO - synthDrivers.espeak.SynthDriver.**init** (07:54:32):
Using eSpeak version 1.45.47  24.Oct.11
INFO - synthDriverHandler.setSynth (07:54:36):
Loaded synthDriver espeak
INFO - core.main (07:54:36):
Using wx version 2.8.11.0 (msw-unicode)
INFO - braille.initialize (07:54:36):
Using liblouis version 2.3.0
INFO - braille.BrailleHandler.setDisplayByName (07:54:36):
Loaded braille display driver noBraille
WARNING - core.main (07:54:38):
Java Access Bridge not available
INFO - core.main (07:54:39):
NVDA initialized
DEBUGWARNING - core.Notify (07:56:25):
errors in this core pump cycle
Traceback (most recent call last):
  File "core.pyo", line 286, in Notify
  File "IAccessibleHandler.pyo", line 852, in pumpAll
  File "IAccessibleHandler.pyo", line 611, in processGenericWinEvent
  File "IAccessibleHandler.pyo", line 533, in winEventToNVDAEvent
  File "NVDAObjects\IAccessible__init__.pyo", line 44, in getNVDAObjectFromEvent
  File "NVDAObjects__init__.pyo", line 78, in **call**
  File "appModules\explorer.pyo", line 91, in chooseNVDAObjectOverlayClasses
  File "baseObject.pyo", line 34, in **get**
  File "baseObject.pyo", line 110, in _getPropertyViaCache
  File "NVDAObjects\IAccessible__init__.pyo", line 877, in _get_parent
  File "NVDAObjects__init__.pyo", line 78, in **call**
  File "appModules\explorer.pyo", line 91, in chooseNVDAObjectOverlayClasses
  File "baseObject.pyo", line 34, in **get**
  File "baseObject.pyo", line 110, in _getPropertyViaCache
  File "NVDAObjects\IAccessible__init__.pyo", line 880, in _get_parent
  File "NVDAObjects__init__.pyo", line 78, in **call**
  File "appModules\explorer.pyo", line 98, in chooseNVDAObjectOverlayClasses
  File "baseObject.pyo", line 34, in **get**
  File "baseObject.pyo", line 110, in _getPropertyViaCache
  File "NVDAObjects\IAccessible__init__.pyo", line 1315, in _get_parent
  File "baseObject.pyo", line 34, in __get__
  File "baseObject.pyo", line 110, in _getPropertyViaCache
  File "NVDAObjects\window__init__.pyo", line 260, in _get_parent
  File "NVDAObjects__init__.pyo", line 65, in **call**
  File "watchdog.pyo", line 147, in _COMError_init
CallCancelled
DEBUGWARNING - core.Notify (07:59:58):
errors in this core pump cycle
Traceback (most recent call last):
  File "core.pyo", line 286, in Notify
  File "IAccessibleHandler.pyo", line 852, in pumpAll
  File "IAccessibleHandler.pyo", line 611, in processGenericWinEvent
  File "IAccessibleHandler.pyo", line 533, in winEventToNVDAEvent
  File "NVDAObjects\IAccessible__init__.pyo", line 44, in getNVDAObjectFromEvent
  File "NVDAObjects__init__.pyo", line 65, in **call**
  File "watchdog.pyo", line 147, in _COMError_init
CallCancelled
DEBUGWARNING - core.Notify (07:59:59):
errors in this core pump cycle
Traceback (most recent call last):
  File "core.pyo", line 286, in Notify
  File "IAccessibleHandler.pyo", line 855, in pumpAll
  File "IAccessibleHandler.pyo", line 747, in processForegroundWinEvent
  File "IAccessibleHandler.pyo", line 525, in winEventToNVDAEvent
  File "_UIAHandler.pyo", line 212, in isUIAWindow
  File "watchdog.pyo", line 140, in sendMessageCallCanceller
CallCancelled
DEBUGWARNING - core.Notify (08:00:07):
errors in this core pump cycle
Traceback (most recent call last):
  File "core.pyo", line 286, in Notify
  File "IAccessibleHandler.pyo", line 838, in pumpAll
  File "IAccessibleHandler.pyo", line 747, in processForegroundWinEvent
  File "IAccessibleHandler.pyo", line 533, in winEventToNVDAEvent
  File "NVDAObjects\IAccessible__init__.pyo", line 44, in getNVDAObjectFromEvent
  File "NVDAObjects__init__.pyo", line 65, in **call**
  File "watchdog.pyo", line 147, in _COMError_init
CallCancelled
DEBUGWARNING - core.Notify (08:11:04):
errors in this core pump cycle
Traceback (most recent call last):
  File "core.pyo", line 286, in Notify
  File "IAccessibleHandler.pyo", line 852, in pumpAll
  File "IAccessibleHandler.pyo", line 611, in processGenericWinEvent
  File "IAccessibleHandler.pyo", line 533, in winEventToNVDAEvent
  File "NVDAObjects\IAccessible__init__.pyo", line 44, in getNVDAObjectFromEvent
  File "NVDAObjects__init__.pyo", line 65, in **call**
  File "watchdog.pyo", line 147, in _COMError_init
CallCancelled
DEBUGWARNING - core.Notify (10:08:11):
errors in this core pump cycle
Traceback (most recent call last):
  File "core.pyo", line 286, in Notify
  File "IAccessibleHandler.pyo", line 852, in pumpAll
  File "IAccessibleHandler.pyo", line 611, in processGenericWinEvent
  File "IAccessibleHandler.pyo", line 533, in winEventToNVDAEvent
  File "NVDAObjects\IAccessible__init__.pyo", line 44, in getNVDAObjectFromEvent
  File "NVDAObjects__init__.pyo", line 55, in **call**
  File "NVDAObjects__init__.pyo", line 182, in findBestAPIClass
  File "NVDAObjects\IAccessible\MSHTML.pyo", line 334, in kwargsFromSuper
  File "watchdog.pyo", line 147, in _COMError_init
CallCancelled
DEBUGWARNING - core.Notify (10:08:12):
errors in this core pump cycle
Traceback (most recent call last):
  File "core.pyo", line 286, in Notify
  File "IAccessibleHandler.pyo", line 838, in pumpAll
  File "IAccessibleHandler.pyo", line 747, in processForegroundWinEvent
  File "IAccessibleHandler.pyo", line 533, in winEventToNVDAEvent
  File "NVDAObjects\IAccessible__init__.pyo", line 44, in getNVDAObjectFromEvent
  File "NVDAObjects__init__.pyo", line 55, in **call**
  File "NVDAObjects__init__.pyo", line 182, in findBestAPIClass
  File "NVDAObjects\IAccessible\MSHTML.pyo", line 334, in kwargsFromSuper
  File "watchdog.pyo", line 147, in _COMError_init
CallCancelled
```

**Comment by bhavyashah** [comment]
I can safely say that this issue hasn't completely disappeared even with version 2017.2, and the results are exactly same as described in many of these comments, but there remains no reliable method of replication. Thoughts/inputs anyone?

**Comment by bhavyashah** [comment]
@feerrenrut While diagnosing this issue may be difficult, I do believe that it has a high impact. NVDA users have been experiencing this problem since years and I am sure we still have testers like me and others who can replicate this from time to time. I would like to thus kindly request technical reconsideration of this ticket as well as the assignment of a priority for the same.

**Comment by feerrenrut** [comment]
@bhavyashah, since this has a long running history, could you summarise this issue?

**Comment by Adriani90** [comment]
@mehgcap,
@bhavyashah,
is this issue still reproducible with NVDA 2018.2.1 on Windows 7 or higher?

**Comment by Adriani90** [comment]
Do the modifier keys work properly with other screen readers?

**Comment by Adriani90** [comment]
I cannot reproduce this issue in NVDA 2018.3.2 in Windows 10 last update. Since this issue was originally reported for Windows xp, could someone report if it is reproducible in Windows 8 or higher?
@bhavyashah, @aleksey_s, @elliott94, @aaronius

**Comment by Adriani90** [comment]
@aliminator

**Comment by fernando-jose-silva** [comment]
I have this problem on a machine.
When windows starts with nvda:
Neither a nvda command button works example nvda + t, nvda + n among others.
The ctrl key does not interrupt speech.
When pressing a script as an example in a list by pressing several arrows below to quickly step through the items, nvda does not interrupt the speech of previous items, repeating the entire command history.
In the log that I attach another point that caught my attention, the log does not record a keystroke, only the speech and errors of the nvda.
Everything returns to normal when restarting the nv
[loknvda1.txt](https://github.com/nvaccess/nvda/files/2546440/loknvda1.txt)
da.
When restarting windows the problem occurs again.

**Comment by Adriani90** [comment]
This is an indicator that another program kills most parts of NVDA except for speech synth. I saw this issue from time to time as well. I can reproduce it by doing following:
1. Open NVDA 2019.1.1
2. Open the attached word dokument
3. Press ctrl+shift + down arrow to select paragraphs and hold it pressed for some seconds

Actual: NVDA is killed almost completely. NVDA key strokes to not work, nvda+q does not work etc. However, the synthesizer still works but does not react on every key press.

Expected: NVDA should not be killed.
For log file and errors related to this issue, see #9463.

**Comment by Adriani90** [comment]
[ONKYO.docx](https://github.com/nvaccess/nvda/files/3135188/ONKYO.docx)

**Comment by Adriani90** [comment]
Another example where NVDA's synthesizer does not react when pressing tab or arrow keys is following:
1. Open a portable copy of NVDA
2. Open CCleaner

Actual: When pressing tab and shift+tab NVDA does not react. imediately
Expected: NVDA should react imediately as it reacts in the installed version.

**Comment by Adriani90** [comment]
for the issue with the attached word document there is a separate one in place. Regarding CCleaner issue, it works properly now in NVDA 2020 RC1, running as administrator.
I am closing this issue because we don't have meaninful updates from the initial author or several other users marked in this issue.
If you have a current use case where you can reproduce this reliably, please open a new issue. Thanks.
